lightning-c-bindings/README.md

```diff
@@ -0,0 +1,128 @@
+The wrapper crate and C/C++ Headers in this folder are auto-generated from the Rust-Lightning
+source by the c-bindings-gen crate contained in the source tree and
+[cbindgen](https://github.com/eqrion/cbindgen). They are intended to be used as a base for building
+language-specific bindings, and are thus incredibly low-level and may be difficult to work with
+directly.
+
+In other words, if you're reading this, you're either writing bindings for a new language, or
+you're in the wrong place - individual language bindings should come with their own documentation.
+
+LDK C Bindings
+===================
+
+The C bindings available at include/lightning.h require inclusion of include/rust_types.h first.
+
+All of the Rust-Lightning types are mapped into C equivalents which take a few forms, namely:
+
+ * Structs are mapped into a simple wrapper containing a pointer to the native Rust-Lightning
+   object and a flag to indicate whether the object is owned or only a reference. Such mappings
+   usually generate a `X_free` function which must be called to release the allocated resources.
```

Comment:
It would be useful to note when an `X_free` function is not generated so it's clear whether one needs to be called.

Reply:
Hmm, I'm not sure we want to commit to such a thing - basically it exists for all types except a few that we know can't need it (maybe only unitary enums?), but not sure if we want make it a "stable" ABI.